core/common/src/main/java/alluxio/worker/block/io/WorkerInternalBlockWorker.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.block.io;
+
+import alluxio.exception.BlockAlreadyExistsException;
+import alluxio.exception.BlockDoesNotExistException;
+import alluxio.exception.InvalidWorkerStateException;
+import alluxio.exception.WorkerOutOfSpaceException;
+import alluxio.grpc.AsyncCacheRequest;
+import alluxio.wire.BlockReadRequest;
+
+import java.io.IOException;
+
+/**
+ * An interface for worker internal clients to interact with worker operations directly
+ * without going through external RPC frameworks.
+ */
+public interface WorkerInternalBlockWorker {
```

Comment:
- Why this is called `WorkerInternalBlockWorker`, the name is confusing
- actually we do we need an interface with a subset of methods of `BlockWorker`?

Reply:
The reason i call it `WorkerInternalBlockWorker` is because i think `EmbeddedBlockWorker` is confusing as well. The code is client code but it calls `Embedded...`, I don't know what it embedded to, so I give this name, any suggestion for the naming?